src/convert_issues_to_pages.py

```diff
@@ -213,186 +190,71 @@ def generate_issue_line(consolidated_issue: ConsolidatedIssue) -> str:
     """
         Generates an index summary markdown page for all features.
 
         @param issue_markdown_content: A string containing all the generated markdown blocks for the features.
         @param template_index_page: The string template for the index markdown page.
-        @param milestones_as_chapters: A boolean switch indicating whether milestones should be treated as chapters.
+        @param output_directory: The directory where the index page will be saved.
 
         @return: None
     """
 
     # Prepare issues replacement for the index page
-    issues_replacement = {
+    replacement = {
+        "date": datetime.now().strftime("%Y-%m-%d"),
         "issues": issue_markdown_content
     }
 
     # Replace the issue placeholders in the index template
-    index_page_content = replace_template_placeholders(template_index_page, issues_replacement)
-
-    # Generate table of contents, if config value of `milestones_as_chapters` is True
-    table_of_contents = generate_table_of_contents(index_page_content) if milestones_as_chapters else ""
-
-    # Prepare additional replacements for the index page
-    replacements = {
-        "date": datetime.now().strftime("%Y-%m-%d"),
-        "table-of-contents": table_of_contents
-    }
-
-    # Second wave of replacing placeholders in the index template
-    index_page = replace_template_placeholders(index_page_content, replacements)
+    index_page = replace_template_placeholders(template_index_page, replacement)
 
     # Create an index page file
-    with open(os.path.join(OUTPUT_DIRECTORY, "_index.md"), 'w', encoding='utf-8') as index_file:
+    with open(os.path.join(output_directory, "_index.md"), 'w', encoding='utf-8') as index_file:
         index_file.write(index_page)
 
-    print("Generated _index.md.")
+    logging.info("Generated _index.md.")
 
 
 def main() -> None:
-    print("Script for converting consolidated issues into md pages started.")
+    # Configure logging
+    logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
+    logging.info("Script for converting consolidated issues into markdown pages started.")
 
-    # Get environment variable, if the milestones should be treated as chapters
-    milestones_as_chapters = os.getenv('MILESTONES_AS_CHAPTERS')
-    milestones_as_chapters = milestones_as_chapters.lower() == 'true'
+    # Load action inputs from the environment and prepare output directory
+    action_inputs = ActionInputs().load_from_environment()
+    output_directory_root = f"{action_inputs.output_directory}/liv-doc"
+    output_directory = os.path.join(output_directory_root, OUTPUT_DIRECTORY_ISSUE)
 
     # Check if the output directory exists and create it if not
     current_dir = os.path.dirname(os.path.abspath(__file__))
-    ensure_folder_exists(os.path.join(current_dir, OUTPUT_DIRECTORY), current_dir)
+    ensure_folder_exists(os.path.join(current_dir, output_directory), current_dir)
```

Comment:
This line looks wrong. See the join here and then join inside.

Reply:
Variable is now set in one line using .join(). Change can be seen in commit: [fb78190](https://github.com/AbsaOSS/living-doc-generator/pull/16/commits/fb78190e5adedc65ded39abff55adc6f51da5407).

Reply:
Here, I mean that `ensure_folder_exists` receives `folder_name` to check its existence in `current_dir.`
But, we are not providing folder name but the path in string.

Reply:
Extra join for absolute paths fixed in commit: [26b1a46](https://github.com/AbsaOSS/living-doc-generator/pull/16/commits/26b1a46fafe63ef4295133e7696547b09fc1ed95)